app/src/main/java/com/github/fribourgsdp/radio/GameActivity.kt

```diff
@@ -41,15 +51,18 @@ class GameActivity : AppCompatActivity(), GameView {
             val hostGameHandler = HostGameHandler(game, this)
             hostGameHandler.linkToDatabase()
             user = User("The best player")
-            playerGameHandler = PlayerGameHandler(game.id, this)
-        } else {
-            playerGameHandler = PlayerGameHandler(intent.getLongExtra(GAME_UID_KEY, -1L), this)
         }
-
+        playerGameHandler = PlayerGameHandler(intent.getLongExtra(GAME_UID_KEY, -1L), this)
         playerGameHandler.linkToDatabase()
 
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+        voiceChannel.leaveChannel()
+        RtcEngine.destroy()
+    }
+
```

Comment:
Yes indeed, as long as the activity isn't destroyed, the players are in the voice chat. This is great because it allows pop-up fragments to appear (when the singer must choose a song) without interrupting the audio.

Reply:
makes sense thank you